src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -775,40 +1007,39 @@ namespace nbl::ext::imgui
-			std::make_pair(bool(m_creationParams.resources.count >= 1u), "Invalid `m_creationParams.resources.count` is equal to 0!"),
-			std::make_pair(bool(m_creationParams.resources.textures.setIx <= 3u), "Invalid `m_creationParams.resources.textures` is outside { 0u, 1u, 2u, 3u } set!"),
-			std::make_pair(bool(m_creationParams.resources.samplers.setIx <= 3u), "Invalid `m_creationParams.resources.samplers` is outside { 0u, 1u, 2u, 3u } set!"),
-			std::make_pair(bool(m_creationParams.resources.textures.bindingIx != m_creationParams.resources.samplers.bindingIx), "Invalid `m_creationParams.resources.textures.bindingIx` is equal to `m_creationParams.resources.samplers.bindingIx`!"),
-			std::make_pair(bool(validateResourcesInfo()), "Invalid `m_creationParams.resources` content!")
+			std::make_pair(bool(creationParams.assetManager), "Invalid `creationParams.assetManager` is nullptr!"),
+			std::make_pair(bool(creationParams.assetManager->getSystem()), "Invalid `creationParams.assetManager->getSystem()` is nullptr!"),
+			std::make_pair(bool(creationParams.utilities), "Invalid `creationParams.utilities` is nullptr!"),
+			std::make_pair(bool(creationParams.transfer), "Invalid `creationParams.transfer` is nullptr!"),
+			std::make_pair(bool(creationParams.renderpass), "Invalid `creationParams.renderpass` is nullptr!"),
+			(creationParams.assetManager && creationParams.utilities && creationParams.transfer && creationParams.renderpass) ? std::make_pair(bool(creationParams.utilities->getLogicalDevice()->getPhysicalDevice()->getQueueFamilyProperties()[creationParams.transfer->getFamilyIndex()].queueFlags.hasFlags(IQueue::FAMILY_FLAGS::TRANSFER_BIT)), "Invalid `creationParams.transfer` is not capable of transfer operations!") : std::make_pair(false, "Pass valid required UI::S_CREATION_PARAMETERS!"),
+			std::make_pair(bool(creationParams.resources.texturesInfo.setIx <= 3u), "Invalid `creationParams.resources.textures` is outside { 0u, 1u, 2u, 3u } set!"),
+			std::make_pair(bool(creationParams.resources.samplersInfo.setIx <= 3u), "Invalid `creationParams.resources.samplers` is outside { 0u, 1u, 2u, 3u } set!"),
+			std::make_pair(bool(creationParams.resources.texturesInfo.bindingIx != creationParams.resources.samplersInfo.bindingIx), "Invalid `creationParams.resources.textures.bindingIx` is equal to `creationParams.resources.samplers.bindingIx`!"),
+			std::make_pair(bool(validateResourcesInfo()), "Invalid `creationParams.resources` content!")
 		});
 
 		for (const auto& [ok, error] : validation)
 			if (!ok)
 			{
-				m_creationParams.utilities->getLogger()->log(error, system::ILogger::ELL_ERROR);
+				creationParams.utilities->getLogger()->log(error, ILogger::ELL_ERROR);
 				assert(false);
```

Comment:
stop it with the assertions, if you need to fail then move stuff from a constructor to a `static core::smart_refctd_ptr<UI> UI::create` or  `static std::unique_ptr<UI> UI::create` method and make the ctor protected/private which only does the "cannot fail stages"

Reply:
resolved